tests/baselines/reference/declarationEmitNoNonRequiredParens.js

```diff
@@ -29,4 +29,4 @@ export declare enum Test {
     C = 2
 }
 export type TestType = typeof Test;
-export declare const bar: Test[];
+export declare const bar: TestType[Extract<keyof TestType, string>][];

```

Comment:
Hm, this test probably needs some kind of extra indirection to prevent node reuse - it definitely wanted to baseline the result of this operation, not copy this operation.

Reply:
Added `satisfies any` which the syntactic printer will not look past